compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -234,7 +250,7 @@ generateTemplateInstance ::
     -> [(LF.TypeVarName, LF.Kind)]
     -> LF.PackageId
     -> HsDecl GhcPs
-generateTemplateInstance env typeCon typeParams externPkgId =
+generateTemplateInstance env dataTypeCon dataParams externPkgId =
```

Comment:
The renaming seems slightly confusing in `data X = Y`, `X` is a type constructor and `Y` is a data constructor. I’ve never heard the term datatype constructor. As for `dataParams` this sounds like it might refer to the fields in a data constructor.

Leaving the naming aside, is there ever a case where `dataParams` is non-empty at the moment? Given that we monomorphize types in DAML-LF, I’m having trouble coming up with a way to hit this. If not, it would be good to document this in a comment (this will most likely change in DAML-LF soonish so no need to remove it completely).

Reply:
The reason for the name change is to match the names used by the only caller of the function.
And at this point, the names are bound by the `LF.DefDataType{..}`
`dataTypeCon` and `dataParams` being the field names of `DefDataType`.

Regarding whether `dataParams` can be non-empty: I'm not sure. This code was pre-existing. So I assume Robin passed this because it could be non-empty.  @robin-da ?

Reply:
It makes some sense in the datatype definition due to Haskell’s issues with colliding record field names. I don’t think it makes any sense here.

Reply:
reverted name change

Reply:
the question about if `dataParams` can be non-empty is still open

Reply:
I’m fine leaving it for now and we can sort it out later.